app/src/main/java/org/mozilla/tv/firefox/utils/BuildConfigDerivables.kt

```diff
@@ -4,18 +4,35 @@
 
 package org.mozilla.tv.firefox.utils
 
+import android.net.Uri
 import org.mozilla.tv.firefox.BuildConfig
 import org.mozilla.tv.firefox.pocket.PocketVideoRepo
 
+private const val POCKET_PARAM_API_KEY = "consumer_key"
+
 /**
  * Computes information that must be derived from the [BuildConfig].
  *
  * This logic is often simple but noisy, so pulling it out of client code improves readability.
  */
 class BuildConfigDerivables {
-    @Suppress("SENSELESS_COMPARISON") // Values of BuildConfig can change but the compiler doesn't know that
+    @Suppress("UselessCallOnNotNull") // Values of BuildConfig can change but the compiler doesn't know that
     val initialPocketRepoState = when {
-        BuildConfig.POCKET_KEY == null -> PocketVideoRepo.FeedState.NoAPIKey
+        BuildConfig.POCKET_KEY.isNullOrEmpty() -> PocketVideoRepo.FeedState.NoAPIKey
         else -> PocketVideoRepo.FeedState.Loading
     }
+
+    // Pocket key can be null if it was not included in the build.  In this case we
+    // know that any calls to Pocket will fail, and so we do not provide the
```

Comment:
So essentially we are preventing the app from executing Pocket endpoint requests. But this also means we are not stopping async request calls in general. Fire stick devices already have perf issues due to its hardware spec

Reply:
That's a good point.  I wanted to cut this off at the narrowed point possible, rather than just turning off updates, because I didn't want us to accidentally write code that bypassed those checks.  This is part of what caused our earlier problem (our initial update makes sure we have a valid key, but our background updates don't).  Killing this at the `Uri` seemed like a good way to avoid that.  

Still, it would probably be good to take a look at this and make sure the performance impact is not significant.  My guess is that it wouldn't be.  It's basically waiting for a few minutes, seeing that there's a null, and then shorting with this code.  But it would be good to verify that I'm not missing anything that could hurt performance.